api/src/main/java/org/eclipse/microprofile/metrics/MetricID.java

```diff
@@ -0,0 +1,176 @@
+/*
+ **********************************************************************
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *               2018 IBM Corporation and others
+ *               and other contributors as indicated by the @author tags.
+ *
+ * See the NOTICES file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ **********************************************************************/
+package org.eclipse.microprofile.metrics;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ *
+ * A unique identifier for {@link Metric} and {@link Metadata} that are registered
+ * in the {@link MetricRegistry}
+ *
+ * The MetricID contains:
+ * <ul>
+ * <li>
+ * {@code Name}: (Required) The name of the metric.
+ * </li>
+ * <li>
+ * {@code Tags}: (Optional) The tags (represented by key/value pairs) of the metric which is augmented by global tags (if available).
+ * Global tags can be set by passing the list of tags in an environment variable {@code MP_METRICS_TAGS}.
+ * For example, the following can be used to set the global tags:
+ *
+ * <pre>
+ * <code>
+ *      export MP_METRICS_TAGS=app=shop,tier=integration
+ * </code>
+ * </pre>
+ *
+ * </li>
+ * </ul>
+ */
+public class MetricID {
```

Comment:
Should we consider making this immutable? `Metadata` is also immutable now. Just making `addTag` and `addTags` methods private would do the job IMO.